cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -593,11 +547,11 @@ void reader::impl::allocate_nesting_info()
     auto cur_schema = _metadata->get_schema(schema_idx);
     int cur_depth   = max_depth - 1;
```

Comment:
Added the check.  Worth pointing out that for historical reasons, `page_nesting_info` is sometimes over-allocated. We alllocate the worst case of max input column depth (ie, depth in the parquet schema tree) and max output column depth (depth in the output cudf columns) for convenience.  `page_nesting_decode_info` needs to remain that way, but `page_nesting_info` could be reduced.   I think that's better done as a followup though, since it's tangential to this PR.

Reply:
So it turns out there is a bug here, but it's very old.  Verified with mainline 24.02.  In some cases (certain schema arragements), the 0th element can remain uninitialized, which has a mild side effect:  The size estimation for output chunks (but only output chunks, not the input stuff in this PR) will end up a little small for list columns.

I'm going to revert this change for now and do a proper fix as a followup so I don't accidentally blow up this PR :)

https://github.com/rapidsai/cudf/issues/14634